endpoints/meme.py

```diff
@@ -0,0 +1,97 @@
+from io import BytesIO
+
+from PIL import ImageDraw
+from flask import send_file
+
+from utils import http
+from utils.endpoint import Endpoint, setup
+
+
+@setup
+class Meme(Endpoint):
+    params = ['avatar0', 'text']
+
+    def generate(self, avatars, text, usernames):
+        img = http.get_image(avatars[0]).convert('RGB')
+        font = self.assets.get_font('assets/fonts/impact.ttf', size=48, )
+        # parse top and bottom text
+        try:
+            top_text, bottom_text = text.replace(', ', ',').split(',')
+        except ValueError:
+            top_text, bottom_text = 'TOP TEXT,BOTTOM TEXT'.split(',')
+        draw = ImageDraw.Draw(img)
+
+        def draw_text_with_outline(text, x, y):
+            draw.text((x - 2, y - 2), text, (0, 0, 0), font=font)
+            draw.text((x + 2, y - 2), text, (0, 0, 0), font=font)
+            draw.text((x + 2, y + 2), text, (0, 0, 0), font=font)
+            draw.text((x - 2, y + 2), text, (0, 0, 0), font=font)
+            draw.text((x, y), text, (255, 255, 255), font=font)
+            return
+
+        def draw_text(string, pos):
```

Comment:
This could probably be separated into "chunks" and moved into utils to measure and fit text.

Reply:
I don't think it's really feasible as this is really long and meme specific, it's probably better to rethink (ha) the wrap method.